docs/learn/beginner/01-tx-lifecycle.md

```diff
@@ -112,6 +112,10 @@ A copy of the cached context is provided to the `AnteHandler`, which performs li
 
 For example, the [`auth`](https://github.com/cosmos/cosmos-sdk/tree/main/x/auth) module `AnteHandler` checks and increments sequence numbers, checks signatures and account numbers, and deducts fees from the first signer of the transaction - all state changes are made using the `checkState`.
 
+:::warning
+Note that ante handlers only run on a transaction. If a transaction embed multiple messages (like some x/authz, x/gov txs for instance), the ante handlers will run on the outer message. Inner messages are mostly directly routed to the [message router](https://docs.cosmos.network/main/learn/advanced/baseapp#msg-service-router) and skip the ante handler. This is something to keep in mind when designing your own ante handler.
```

Comment:
> ... the ante handlers will run on the outer message.

I would perhaps consider rephrasing this. I think what you mean to say is that the AnteHandler chain has context or awareness of only the outer messages, not that it runs or processes them. I just want to avoid ambiguity or confusion.